### DISCUSSION TITLE
Do not run cron workflows in forks

### DISCUSSION THREAD

**Discussion by jacobtomlinson** [community/community#26684] [Actions]
<p>I have an action which runs on a cron schedule. However I only want it to run on the main repo, not in forks. Currently it runs in all forks too.</p>
<p>Is there a way to limit the workflow to a specific repo name, similar to how you can filter on modified files?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jacobtomlinson">@jacobtomlinson</a> ,</p>
<p>In the workflow, you can use the property <strong>github.repository</strong> of <a href="https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a>, or the <strong><a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">environment variable</a> GITHUB_REPOSITORY</strong>, to view the repository where the current workflow is running on.</p>
<p>You can use the <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener"><strong>if conditional</strong></a> to skip all the jobs of the scheduled workflow according to the repository.</p>
<p>For example, the main repository is _ <strong>main_org/main_repo</strong> _:</p>
<pre><code>Jobs:
  build:
    if: (github.event_name == 'schedule' &amp;&amp; github.repository == 'main_org/main_repo') || (github.event_name != 'schedule')
    . . .
</code></pre>
<p>This workflow will execute the job _ <strong>build</strong> _ only when meeting one of the following situations:</p>
<ol>
<li>
<p>The event is _ <strong>schedule</strong> <em> and the repository is </em> <strong>main_org/main_repo</strong> <em>, if the repository is not </em> <strong>main_org/main_repo</strong> _ the job _ <strong>build</strong> _ will be skipped.</p>
</li>
<li>
<p>The event is not _ <strong>schedule </strong> _and the repository can be anyone.</p>
</li>
</ol>
<p>You just need to setup this configuration once on the main repository, and every time you create a new fork from the main repository, this configuration will be copy to the fork repository.</p>
<p>Of course, you also can try to remove the _ <strong>schedule</strong> _ event and only retain the needed events in the YAML file from the fork repository. However, if you need to create multiple fork repositories, this may become more troublesome.</p>

**Comment by jacobtomlinson** [comment]
<p>Perfect thanks <a class="mention" href="/brightran">@brightran</a>! The workflow raises a PR automatically to update dependencies. So raising this PR on forks was just noise for anyone who contributes to the project.</p>
<p>Adding the following resolved it for me:</p>
<pre><code>jobs:
  job-name:
    if: github.repository == 'org_name/repo_name'

steps:
...
</code></pre>

**Comment by mkurz** [comment]
<p>Checking for the repository name on the job level is a bit cumbersome IMHO.<br>
I opened a ticket with suggestions of better solutions: <a href="https://github.com/github/feedback/discussions/16109" class="inline-onebox" rel="noopener nofollow ugc">Restrict scheduled workflow to repository · Discussion #16109 · github/feedback · GitHub</a></p>
